src/java.desktop/share/classes/java/awt/geom/AffineTransform.java

```diff
@@ -3902,15 +3902,25 @@ public Object clone() {
      * @since 1.2
      */
     public int hashCode() {
-        long bits = Double.doubleToLongBits(m00);
-        bits = bits * 31 + Double.doubleToLongBits(m01);
-        bits = bits * 31 + Double.doubleToLongBits(m02);
-        bits = bits * 31 + Double.doubleToLongBits(m10);
-        bits = bits * 31 + Double.doubleToLongBits(m11);
-        bits = bits * 31 + Double.doubleToLongBits(m12);
+        long bits = hash(m00);
+        bits = bits * 31 + hash(m01);
+        bits = bits * 31 + hash(m02);
+        bits = bits * 31 + hash(m10);
+        bits = bits * 31 + hash(m11);
+        bits = bits * 31 + hash(m12);
         return (((int) bits) ^ ((int) (bits >> 32)));
     }
 
+    /**
+     * Returns a hash code for the given value, with negative zero
+     * collapsed into to the single positive zero.
+     */
+    private static long hash(double m) {
+        long h = Double.doubleToLongBits(m);
+        if (h == 0x8000000000000000L) h = 0;    // Replace -0 by +0.
```

Comment:
Another idiom to accomplish this is 

return Double.doubleToLongBits(m + 0.0); // Turn -0.0 to +0.0